src/components/Events.ts

```diff
@@ -1,20 +1,22 @@
 import {Component} from 'react';
-import {Navigation, NavigationComponentListener} from 'react-native-navigation';
+import {EventSubscription, Navigation, NavigationComponentListener} from 'react-native-navigation';
 
 export type EventsListener = NavigationComponentListener;
 
 export default class Events {
   originComponentId: string;
+  eventSubscription: EventSubscription | null;
```

Comment:
What's the point in keeping the eventSubscription member? 
Are you using it anywhere? 

Also you set it in two different methods, so basically you override it if you call both methods.

Reply:
1. On functional components, you may register the events on `useEffect` (as happen in `wix-one-app-custodia-cc`) and then to use `remove()` on the cleanup.
2. Basically, you're not suppose to use these both methods together because one is an extension of the other. But, would you suggest to do it differently?

Reply:
Ok, if you are certain that no one should use both then I guess it's fine. 
Anyway, I would encapsulate the `this.eventSubscription` and instead maybe provide a `clean` method that does the cleanup for you. 
If in the future we'll change the way we register events internally we might want to change the cleanup as well.